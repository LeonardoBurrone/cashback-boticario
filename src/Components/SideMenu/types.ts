export type MenuOption = {
  description: string;
  icon: string;
  path: string;
  title: string;
};
