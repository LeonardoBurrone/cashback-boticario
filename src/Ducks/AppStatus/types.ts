export enum AppStatusActionTypes {
  CHANGE_IS_LOGGED_IN = 'CHANGE_IS_LOGGED_IN',
  CHANGE_LOADING = 'CHANGE_LOADING',
  CHANGE_LOADING_MESSAGE = 'CHANGE_LOADING_MESSAGE'
}

export type AppStatusState = {
  isLoggedIn: boolean;
  loading: boolean;
  loadingMessage: string;
};

export type ChangeIsLoggedInAction = {
  payload: boolean;
  type: typeof AppStatusActionTypes.CHANGE_IS_LOGGED_IN;
};

export type ChangeLoadingAction = {
  payload: boolean;
  type: typeof AppStatusActionTypes.CHANGE_LOADING;
};

export type ChangeLoadingMessageAction = {
  payload: string;
  type: typeof AppStatusActionTypes.CHANGE_LOADING_MESSAGE;
};
