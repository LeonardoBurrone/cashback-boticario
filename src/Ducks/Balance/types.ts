export enum BalanceActionTypes {
  CHANGE_BALANCE = 'CHANGE_BALANCE',
  CHANGE_BALANCE_ERROR = 'CHANGE_BALANCE_ERROR',
  FETCH_BALANCE = 'FETCH_BALANCE'
}

export type BalanceState = {
  balance: string;
  balanceError: boolean;
};

export type ChangeBalanceAction = {
  payload: string;
  type: typeof BalanceActionTypes.CHANGE_BALANCE;
};

export type ChangeBalanceErrorAction = {
  payload: boolean;
  type: typeof BalanceActionTypes.CHANGE_BALANCE_ERROR;
};

export type FetchBalanceAction = {
  type: typeof BalanceActionTypes.FETCH_BALANCE;
};

export type BalanceResponse = {
  balance: string;
};
