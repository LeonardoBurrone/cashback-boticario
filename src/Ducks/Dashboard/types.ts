export enum DashboardActionTypes {
  CHANGE_PURCHASES = 'CHANGE_PURCHASES',
  FETCH_PURCHASES = 'FETCH_PURCHASES',
  FETCH_PURCHASES_ERROR = 'FETCH_PURCHASES_ERROR'
}

export type Purchase = {
  cashbackPercentage: number;
  cashbackValue: number;
  date: string;
  id: number;
  price: number;
  status: 'Approved' | 'Rejected' | 'Validating';
};

export type DashboardState = {
  fetchPurchasesError: boolean;
  purchases: Purchase[];
};

export type ChangePuchasesAction = {
  payload: Purchase[];
  type: typeof DashboardActionTypes.CHANGE_PURCHASES;
};

export type FetchPuschasesAction = {
  type: typeof DashboardActionTypes.FETCH_PURCHASES;
};

export type FetchPuschasesErrorAction = {
  payload: boolean;
  type: typeof DashboardActionTypes.FETCH_PURCHASES_ERROR;
};

export type PurchasesResponse = {
  purchases: Purchase[];
};
