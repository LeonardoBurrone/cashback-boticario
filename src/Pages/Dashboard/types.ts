import { Purchase } from '../../Ducks/Dashboard/types';

export type MockState = {
  balance: {
    balance: string;
    balanceError: boolean;
  };
  dashboard: {
    fetchPurchasesError: boolean;
    purchases: Purchase[];
  };
};
