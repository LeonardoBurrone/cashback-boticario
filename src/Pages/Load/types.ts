export type MockState = {
  appStatus: {
    loading: boolean;
    loadingMessage: string;
    requestFailed: boolean;
  };
};
