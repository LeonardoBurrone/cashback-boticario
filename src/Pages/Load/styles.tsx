import { Dialog } from '@material-ui/core';
import styled from 'styled-components';

const CBDialog = styled(Dialog)`
  & .MuiDialog-paperScrollPaper {
    align-items: center;
    padding: 30px;
  }
`;

export { CBDialog };
