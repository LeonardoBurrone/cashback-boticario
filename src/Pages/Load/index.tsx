import { CircularProgress, Typography } from '@material-ui/core';
import { useTheme } from '@material-ui/core/styles';
import React from 'react';
import { useSelector } from 'react-redux';

import { CBDialog } from './styles';

const Load: React.FunctionComponent = () => {
  const loading = useSelector((state: reducers.rootReducer) => state.appStatus.loading);
  const loadingMessage = useSelector((state: reducers.rootReducer) => state.appStatus.loadingMessage);
  const theme = useTheme();

  return (
    <CBDialog aria-labelledby="customized-dialog-title" open={loading} theme={theme}>
      <Typography gutterBottom>{loadingMessage}</Typography>
      <CircularProgress />
    </CBDialog>
  );
};

export default Load;
