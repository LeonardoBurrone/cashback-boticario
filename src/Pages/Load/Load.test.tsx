import { Button, CircularProgress, Typography } from '@material-ui/core';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';

import Load from './';
import { MockState } from './types';

import { AppStatusActionTypes } from '../../Ducks/AppStatus/types';

const appStateMock: MockState = {
  appStatus: {
    loading: false,
    loadingMessage: '',
    requestFailed: false
  }
};
const mockDispatch = jest.fn();

jest.mock('react-redux');

describe('Test page Load', () => {
  let wrapper: ShallowWrapper;

  const mockUseSelector = (newState?: MockState) => {
    (useSelector as jest.Mock).mockImplementation((callback: (state: MockState) => void) =>
      callback({ ...appStateMock, ...newState })
    );
  };

  beforeEach(() => {
    (useDispatch as jest.Mock).mockReturnValue(mockDispatch);

    mockUseSelector();

    wrapper = shallow(<Load />);
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('Check if component rendered correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Validate layout when there is a request being made', () => {
    mockUseSelector({
      appStatus: {
        loading: true,
        loadingMessage: 'Fetching data...',
        requestFailed: false
      }
    });

    wrapper = shallow(<Load />);

    expect(wrapper.find(Typography).at(0).prop('children')).toBe('Fetching data...');
    expect(wrapper.find(CircularProgress).exists()).toBeTruthy();
  });

  it('Validate layout when there was a failed request', () => {
    mockUseSelector({
      appStatus: {
        loading: false,
        loadingMessage: 'Error fetching data',
        requestFailed: true
      }
    });

    wrapper = shallow(<Load />);

    expect(wrapper.find(Typography).at(0).prop('children')).toBe('Error fetching data');
    expect(wrapper.find(CircularProgress).exists()).toBeFalsy();

    const button: any = wrapper.find(Button).props();
    button.onClick();

    expect(mockDispatch).toBeCalledWith({
      payload: false,
      type: AppStatusActionTypes.CHANGE_REQUEST_FAILED
    });
  });
});
