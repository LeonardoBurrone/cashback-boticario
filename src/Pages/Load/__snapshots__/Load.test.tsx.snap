// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test page Load Check if component rendered correctly 1`] = `
<Styled(WithStyles(ForwardRef(Dialog)))
  aria-labelledby="customized-dialog-title"
  open={false}
>
  <WithStyles(ForwardRef(Typography))
    gutterBottom={true}
    variant="h6"
  />
  <WithStyles(ForwardRef(CircularProgress)) />
</Styled(WithStyles(ForwardRef(Dialog)))>
`;
